pandas/core/indexes/api.py

```diff
@@ -146,7 +146,8 @@ def _union_indexes(indexes, sort=True):
     if len(indexes) == 1:
         result = indexes[0]
         if isinstance(result, list):
-            result = Index(sorted(result))
+            result = sorted(result) if sort else result
```

Comment:
Can this branch be removed altogether or does that break 35 compatibility?

Reply:
Yeah, it breaks python 3.5 compat.

Reply:
Why not just move the compat check from the construction module to here then? Would be simpler

Reply:
It would be a little more complicated here since we would have to do another compat check in the `len(indexes) > 1` case